File: tools/gensincoeff.py
import math
import sys

def taylor_series(n):
    return [1.0 / math.factorial(2 * i + 1) 
            if i % 2 == 0 else -1.0 / math.factorial(2 * i + 1)
            for i in range(0, n)]

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print("Usage: " + sys.argv[0] + " n")
        sys.exit(1)
    
    n = int(sys.argv[1])

    for num in taylor_series(n):
        print(num)

File: tools/gensincoeff.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <stdint.h>

float factorial(int n)
{
	int i;
	float res = 1;

	for (i = 2; i <= n; i++) {
		res *= i;
	}

	return res;
}


int main(int argc, char *argv[])
{
	int i, n;
	float coeff, fact;
	uint32_t bits;

	if (argc < 2) {
		fprintf(stderr, "Usage: %s n\n", argv[0]);
		exit(EXIT_FAILURE);
	}

	n = atoi(argv[1]);

	for (i = 0; i < n; i++) {
		fact = factorial(2 * i + 1);
		coeff = (i % 2 == 0) ? 1.0 / fact : -1.0 / fact;
		memcpy(&bits, &coeff, sizeof(bits));
		printf("32'h%.8x\n", bits);
	}

	return 0;
}
